### PR TITLE
Add rosdep - changes for vinca

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Three changes:
- Remove the "unix" key that matched both osx and linux, as this cannot be easily handled by rosdep (as far as I know)
- Rename lookup from conda-forge to robostack
- Add ROS_OS_OVERRIDE to the activate scripts. It sets the OS to robostack and the version to the actual platform - this is somewhat misusing the version specifier, but in my opinion it's easiest to get what we want ;).

### CODE REVIEW

File: vinca/resolve.py [hunk 1]
```diff
@@ -30,16 +30,11 @@ def resolve_pkgname_from_indexes(pkg_shortname, conda_index):
     for i in conda_index:
         if pkg_shortname in i:
             sys_platform = map_platform_python_to_conda[config.selected_platform]
-            if "conda-forge" in i[pkg_shortname].keys():
-                if sys_platform in i[pkg_shortname]["conda-forge"]:
-                    return i[pkg_shortname]["conda-forge"][sys_platform]
-                elif "unix" in i[pkg_shortname]["conda-forge"] and sys_platform in [
```
Comment (wolfv) [inline vinca/resolve.py:36]: I think the ability to use `unix` went missing here :)

Comment (Tobias-Fischer) [reply vinca/resolve.py:36]: Deliberately! I don't think that rosdep can handle these cases, hence I removed it. In the ros-noetic PR I manually replaced all "unix" with "linux" + "osx" entries. There were only ~10.

File: vinca/templates/activate.sh.in [hunk 2]
```diff
@@ -14,5 +14,10 @@ else
     source $CONDA_PREFIX/setup.sh
 fi
 
+case "$OSTYPE" in
```
Comment (wolfv) [inline vinca/templates/activate.sh.in:17]: I am unsure if OSTYPE very portable ... haven't seen it used often. 

Maybe we should use `uname` which should give `$(uname) == Darwin` and `$(uname) == Linux`. But idk, maybe OSTYPE is good.

File: vinca/resolve.py (hunk 1 above)
Comment (wolfv) [reply vinca/resolve.py:36]: ok, fine! :)

File: vinca/templates/activate.sh.in (hunk 2 above)
Comment (Tobias-Fischer) [reply vinca/templates/activate.sh.in:17]: It works (at least) in sh, zsh, bash & dash. Should cover most Unix systems?

Comment (wolfv) [reply vinca/templates/activate.sh.in:17]: alright, let's keep it. also just tried on Mac with Bash & zsh.

**Pull request merged**